.gitignore

```diff
@@ -25,3 +25,6 @@ __pycache__
 
 # Emacs backup
 *~
+/.caches/
+/eclipse-build/
```

Comment:
@dsmiley I don't use Eclipse so I don't know if we need this or not, can you PTAL?

Reply:
PTAL?
I don't use Eclipse either.  Let's not modify .gitignore in this PR.

These paths are for previous branches (branch_8x), but not master.  I don't think we should pollute the .gitignore of master branch with directories that should be ignored of previous release branches.  The contributor probably switched branches.  Different checkouts, ideally aided by `git worktree` should be used.

Reply:
Maybe a comment at the bottom of this file could raise awareness of this to anyone who is tempted to touch it?